### PR TITLE
serverless: configure resource limits to git import / s2i / deploy image

### PR DESCRIPTION
**Pull request by gijohn**
Create / Configure resource limits while git import / S2I / Deploy Image

User Story: https://jira.coreos.com/browse/ODC-1323
~~Dependent on PR: https://github.com/openshift/console/pull/1823~~

image

### CODE REVIEW

File: frontend/packages/dev-console/src/components/formik-fields/ResourceLimitField.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import * as React from 'react';
```
Comment (gijohn) [inline frontend/packages/dev-console/src/components/formik-fields/ResourceLimitField.tsx]: from PR https://github.com/openshift/console/pull/1823

File: frontend/packages/dev-console/src/components/formik-fields/field-types.ts [hunk 2]
```diff
@@ -48,3 +48,15 @@ export interface SecretKeyRef {
     name: string;
   };
 }
+
```
Comment (gijohn) [inline frontend/packages/dev-console/src/components/formik-fields/field-types.ts]: from PR https://github.com/openshift/console/pull/1823

File: frontend/packages/dev-console/src/components/formik-fields/index.ts [hunk 3]
```diff
@@ -7,3 +7,4 @@ export { default as EnvironmentField } from './EnvironmentField';
 export { default as DroppableFileInputField } from './DroppableFileInputField';
 export { default as InputSearchField } from './InputSearchField';
 export { default as MultiColumnField } from './multi-column-field/MultiColumnField';
+export { default as ResourceLimitField } from './ResourceLimitField';
```
Comment (gijohn) [inline frontend/packages/dev-console/src/components/formik-fields/index.ts]: from PR https://github.com/openshift/console/pull/1823

File: frontend/packages/dev-console/src/components/import/ImportForm.tsx [hunk 4]
```diff
@@ -94,6 +94,20 @@ const ImportForm: React.FC<ImportFormProps & StateProps> = ({
       replicas: 1,
     },
     labels: {},
+    limits: {
```
Comment (gijohn) [inline frontend/packages/dev-console/src/components/import/ImportForm.tsx]: from PR https://github.com/openshift/console/pull/1823

File: frontend/packages/dev-console/src/components/import/__tests__/import-validation-utils.spec.ts [hunk 5]
```diff
@@ -73,5 +73,39 @@ describe('ValidationUtils', () => {
         );
       });
     });
+
```
Comment (gijohn) [inline frontend/packages/dev-console/src/components/import/__tests__/import-validation-utils.spec.ts]: from PR https://github.com/openshift/console/pull/1823

File: frontend/packages/dev-console/src/components/import/advanced/AdvancedSection.tsx [hunk 6]
```diff
@@ -8,6 +8,7 @@ import ScalingSection from './ScalingSection';
 import ServerlessScalingSection from './ServerlessScalingSection';
 import BuildConfigSection from './BuildConfigSection';
 import DeploymentConfigSection from './DeploymentConfigSection';
+import ResourceLimitSection from './ResourceLimitSection';
```
Comment (gijohn) [inline frontend/packages/dev-console/src/components/import/advanced/AdvancedSection.tsx]: from PR https://github.com/openshift/console/pull/1823

File: frontend/packages/dev-console/src/components/import/advanced/ResourceLimitSection.tsx [hunk 7]
```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
```
Comment (gijohn) [inline frontend/packages/dev-console/src/components/import/advanced/ResourceLimitSection.tsx]: from PR https://github.com/openshift/console/pull/1823

File: frontend/packages/dev-console/src/components/import/import-validation-utils.ts [hunk 8]
```diff
@@ -94,6 +95,84 @@ export const validationSchema = yup.object().shape({
       .string()
       .matches(pathRegex, { message: 'Path must start with /.', excludeEmptyString: true }),
   }),
+  limits: yup.object().shape({
```
Comment (gijohn) [inline frontend/packages/dev-console/src/components/import/import-validation-utils.ts]: from PR https://github.com/openshift/console/pull/1823

File: frontend/packages/dev-console/src/components/import/import-submit-utils.ts [hunk 9]
```diff
@@ -139,6 +140,20 @@ export const createDeploymentConfig = (
               image: `${name}:latest`,
               ports,
               env,
+              resources: {
+                ...((cpu.limit || memory.limit) && {
+                  limits: {
+                    ...(cpu.limit && { cpu: `${cpu.limit}${cpu.limitUnit}` }),
```
Comment (invincibleJai) [inline frontend/packages/dev-console/src/components/import/import-submit-utils.ts]: have you validated with various units or we need some conversion before applying?

Comment (gijohn) [reply frontend/packages/dev-console/src/components/import/import-submit-utils.ts]: I've tried different combinations. It works.

**Pull request merged**